py/nlr.h

```diff
@@ -89,24 +103,26 @@
     #define MICROPY_NLR_MIPS (1)
     #define MICROPY_NLR_NUM_REGS (MICROPY_NLR_NUM_REGS_MIPS)
 #else
-    #define MICROPY_NLR_SETJMP (1)
-    //#warning "No native NLR support for this arch, using setjmp implementation"
-#endif
+    #if defined(MICROPY_NLR_SETJMP)
+        #error "No native NLR support for this arch, and setjmp was disabled."
+    #else
+        // Enable the fall-back setjump implementation.
+        #define MICROPY_NLR_SETJMP (1)
+        // #warning "No native NLR support for this arch, using setjmp implementation"
+    #endif
 #endif
-
-// *FORMAT-ON*
+#endif  // !MICROPY_NLR_SETJMP
 
 #if MICROPY_NLR_SETJMP
 #include <setjmp.h>
 #endif
 
-typedef struct _nlr_buf_t nlr_buf_t;
-struct _nlr_buf_t {
+typedef struct _nlr_buf_t {
```

Comment:
just bikeshedding, but I thought these typedefs were more readable the way they were

Reply:
I figured it was improving consistency with how we were doing this elsewhere, e.g. https://github.com/micropython/micropython/blob/master/py/runtime.h#L66 and https://github.com/micropython/micropython/blob/master/py/pairheap.h#L48 (amongst others). (But I also like this way better, so ... I'm biased)